### ISSUE TITLE
compound primary key and inheritance

### ISSUE THREAD

**Issue by dkirchhof** [typeorm/typeorm#2865]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.7`

Hello,  
i came accross a problem when using inheritance. It seems that a compound primary key is not generated if one column is defined in a parent and one in a child class.  
Example:
```ts
// no decorator
export abstract class RatingBase
{
    @PrimaryColumn()
    public userId: string;

    @Column()
    public value: number;
}

@Entity("course_ratings")
export class CourseRating extends RatingBase
{
    @PrimaryColumn()
    public courseId: string;
}
```

#15

**Comment by eddieajau** [comment]
I don't think this is a bug because I think this is an example of where it is not appropriate to use inheritance. You could use `implements RatingBase` to ensure consistency, but I wouldn't extend models like this (been there, done that, refactored the t-shirt).

**Comment by dkirchhof** [comment]
Why wouldn't you extend models like this? I don't want to have much duplicated code (there are more models which extends RatingBase) and this is just a small example. There are more attributes in the base class.  
Most of my models also inherit from a BaseEntity class, where id, createdAt, modifiedAt etc. is defined.
